core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -88,15 +89,26 @@ func (r *EvmRegistry) feedLookup(ctx context.Context, checkResults []ocr2keepers
 	lggr := r.lggr.With("where", "FeedLookup")
 	lookups := map[int]*FeedLookup{}
 	for i, res := range checkResults {
-		if res.IneligibilityReason != uint8(UpkeepFailureReasonTargetCheckReverted) {
+		if res.IneligibilityReason != uint8(encoding.UpkeepFailureReasonTargetCheckReverted) {
+			// Feedlookup only works when upkeep target check reverts
 			continue
 		}
 
 		block := big.NewInt(int64(res.Trigger.BlockNumber))
 		upkeepId := res.UpkeepID
 
-		opts := r.buildCallOpts(ctx, block)
+		// Try to decode the revert error into feed lookup format. User upkeeps can revert with any reason, see if they
+		// tried to call mercury
+		lggr.Infof("upkeep %s block %d trying to decodeFeedLookup performData=%s", upkeepId, block, hexutil.Encode(checkResults[i].PerformData))
+		lookup, err := r.decodeFeedLookup(res.PerformData)
+		if err != nil {
+			lggr.Warnf("upkeep %s block %d decodeFeedLookup failed: %v", upkeepId, block, err)
+			// Not feed lookup error, nothing to do here
```

Comment:
this means any revert other than `FeedLookup` will have reason 3 and state 0, not eligible and not retryable